cpp/dolfinx/la/PETScMatrix.cpp

```diff
@@ -190,8 +191,9 @@ PETScMatrix::set_fn(Mat A, InsertMode mode)
 {
   return [A, mode, cache = std::vector<PetscInt>()](
              std::int32_t m, const std::int32_t* rows, std::int32_t n,
-             const std::int32_t* cols, const PetscScalar* vals) mutable {
-    PetscErrorCode ierr;
+             const std::int32_t* cols, const PetscScalar* vals) mutable
+  {
+    [[maybe_unused]] PetscErrorCode ierr;
```

Comment:
Change to exceptions.

Reply:
@garth-wells  I've now changed quite a few of them. Do you want any more of them to be changed?